src/interpolation_caches.jl

```diff
@@ -11,10 +12,16 @@ Extrapolation extends the last linear polynomial on each side.
 
 ## Keyword Arguments
 
-  - `extrapolate`: boolean value to allow extrapolation. Defaults to `false`.
+  - `extrapolation`: The extrapolation type applied left and right of the data. Possible options
+    are `ExtrapolationType.none` (default), `ExtrapolationType.constant`,
+    `ExtrapolationType.linear` and `ExtrapolationType.extension`.
+  - `extrapolation_left`: The extrapolation type applied left of the data. See `extrapolation` for
```

Comment:
Maybe mention, if `extrapolation` is provided, `extrapolation_left` and `extrapolation_right` is neglected.